src/common.js

```diff
@@ -8,14 +8,15 @@ const input = {
   post: core.getMultilineInput('post', {required: true}),
   workingDirectory: core.getInput('working-directory'),
   shell: core.getInput('shell'),
+  postShell: core.getInput('post-shell'),
 };
 
 export async function run() {
-  return runCommands(joinMultilineCommands(input.run))
+  return runCommands(joinMultilineCommands(input.run), input.shell)
 }
 
 export async function post() {
-  return runCommands(joinMultilineCommands(input.post))
+  return runCommands(joinMultilineCommands(input.post), input.postShell)
```

Comment:
backward compatibility is broken if someone has already redefined shell. In this case, the post-shell will start as a `bash` after the upgrade.

Reply:
@jetexe Do you have an option on how to solve it without a new major version?

Reply:
Good point, how about we switch this around?
I.E. `run-shell` rather than `post-shell`

Or we could have `post-shell` be undefined by default and if undefined, it would be populated with the value of `shell`.

Reply:
@hummeltech nice idea! `post-shell` is undefined (empty) by default and is not required. Later we can use the value from the `shell` by default, and override it only if the `post-shell` was provided. @jetexe agreed?